test/Spec.hs

```diff
@@ -0,0 +1,195 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module Main (main) where
+
+import Data.ByteString.Short as SBS
+import Data.Coerce
+import Data.Int
+import Data.Typeable
+import Data.Word
+import Foreign.C.Types
+import Numeric.Natural (Natural)
+import System.Random
+import Test.SmallCheck.Series as SC
+import Test.Tasty
+import Test.Tasty.HUnit
+import Test.Tasty.SmallCheck as SC
+
+#include "HsBaseConfig.h"
+
+import qualified Spec.Range as Range
+import qualified Spec.Run as Run
+
+main :: IO ()
+main =
+  defaultMain $
+  testGroup
+    "Spec"
+    [ floatingSpec @Double
+    , floatingSpec @Float
+    , floatingSpec @CDouble
+    , floatingSpec @CFloat
+    , integralSpec @Word8
+    , integralSpec @Word16
+    , integralSpec @Word32
+    , integralSpec @Word64
+    , integralSpec @Word
+    , integralSpec @Int8
+    , integralSpec @Int16
+    , integralSpec @Int32
+    , integralSpec @Int64
+    , integralSpec @Int
+    , integralSpec @Char
+    , integralSpec @Bool
+    , integralSpec @CBool
+    , integralSpec @CChar
+    , integralSpec @CSChar
+    , integralSpec @CUChar
+    , integralSpec @CShort
+    , integralSpec @CUShort
+    , integralSpec @CInt
+    , integralSpec @CUInt
+    , integralSpec @CLong
+    , integralSpec @CULong
+    , integralSpec @CPtrdiff
+    , integralSpec @CSize
+    , integralSpec @CWchar
+    , integralSpec @CSigAtomic
+    , integralSpec @CLLong
+    , integralSpec @CULLong
+    , integralSpec @CIntPtr
+    , integralSpec @CUIntPtr
+    , integralSpec @CIntMax
+    , integralSpec @CUIntMax
+    , integralSpec @Integer
+    , integralSpec @Natural
+    -- , bitmaskSpec @Word8
+    -- , bitmaskSpec @Word16
+    -- , bitmaskSpec @Word32
+    -- , bitmaskSpec @Word64
+    -- , bitmaskSpec @Word
```

Comment:
Maybe remove commented lines?

Reply:
Yeah, that's an old spec. The code for the spec itself isn't even in the repo anymore. I made the suggestion to delete those lines here: https://github.com/idontgetoutmuch/random/pull/147#discussion_r436491294